pkg/machines/libvirt-common.js

```diff
@@ -780,6 +782,7 @@ export function parseOsInfoList(dispatch, osList) {
     });
 
     dispatch(updateOsInfoList(parsedList));
+    $('#app').attr("osinfo-list-fetched", true);
```

Comment:
Makes sense.

Reply:
I was just changing this and I came up with the following thought.

The osinfolist which we are waiting to get fetched is coming from libosinfo.
We can't expect all users to have libosinfo, since it's in the Recommends section in the spec file, and not in the Requires.
I will remove/disable Create button till the OS list if fetched, what would happen for the users without libosinfo is that they will stay with a disabled Create button without any hint why. And well, the Create modal is useful even without libosinfo.

So, I having seconds thought regarding waiting for the OS list to get fetched before enabling the button.

I prefer to just ignore this. And keep the code in the test checking for the DOM.

Anyway, fetching OS list is very fast, it's just the test that is also very fast and races with the data initialization.

Reply:
If libosinfo is not installed, then the osinfolist promise should come back with a rejection or failure, at which point the button gets enabled, too. I. e. the point is not that it finishes *successfully*, but that it finishes *at all*, either way. There shouldn't ever be a situation where the promise takes indefinitely long?

Reply:
That said, I don't insist on this, it was just a thought to make this a little nicer.

Reply:
Actually I followed your advice, it's indeed nicer.